uisdk/src/main/java/com/karhoo/uisdk/screen/booking/booking/bookingrequest/BookingRequestPresenter.kt

```diff
@@ -150,8 +150,10 @@ class BookingRequestPresenter(view: BookingRequestMVP.View,
     }
 
     override fun setBookingEnablement(hasValidPaxDetails: Boolean) {
-        if (KarhooUISDKConfigurationProvider.configuration.authenticationMethod() is AuthenticationMethod.KarhooUser
-                || (hasValidPaxDetails && userStore.savedPaymentInfo != null)) {
+        if (userStore.savedPaymentInfo == null) {
+            view?.disableBooking()
```

Comment:
They have to enter their payment details, or do you mean we need to notify them? It might be on initial entry to the form e.g. for any Adyen users and for Braintree guest and possible token users.